### PR TITLE
Add doxygen documentation for http demos

### PR DESCRIPTION
**Pull request by yourslab**
As part of this PR, I also split up each library with at least one demo into its own @page along with moving each individual demo as a @subsection.

By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: docs/doxygen/demos/shadow_demo.dox [hunk 1]
```diff
@@ -0,0 +1,32 @@
+
+/**
```
Comment (sarenameas) [inline docs/doxygen/demos/shadow_demo.dox:2]: I like this organization thanks

File: docs/doxygen/demos/http_demo.dox [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+plaintext connection over port 80 to an HTTP server. However, if a connection 
+failure occurs, then the connection establishment is retried for @ref MAX_RETRY_ATTEMPTS.
+Once a connection is established, HTTP requests are sent for the following HTTP methods:
+`GET`, `HEAD`, `PUT`, and `POST`. The respective responses from each of these 
+requests are logged.
+</p>
+
+@image html http_demo_plaintext.png width=100%
+<div class="caption" style="text-align:center">
+HTTP Plaintext Demo Workflow — Note that these steps are repeated indefinitely
```
Comment (sarenameas) [inline docs/doxygen/demos/http_demo.dox:20]: Should the title of the workflow go above the image?

File: docs/doxygen/demos/http_demo.dox [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+</div>
+
+@section http_demo_basic_tls HTTP Basic TLS Demo
+@brief A demo of the coreHTTP library that establishes a TLS session on top of a TCP connection
```
Comment (sarenameas) [inline docs/doxygen/demos/http_demo.dox:24]: Maybe we can differentiate this description from the mutual auth one like the MQTT demo description does?
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/docs/doxygen/demos/demos_subpages.dox#L3

File: docs/doxygen/demos/http_demo.dox [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+requests are logged.
+</p>
+
+@image html http_demo_basic_tls.png width=100%
+<div class="caption" style="text-align:center">
+HTTP Basic TLS Demo Workflow — Note that these steps are repeated indefinitely
+</div>
+
+@section http_demo_mutual_auth HTTP Mutual Auth Demo
+@brief A demo of the coreHTTP library that establishes a TLS session on top of a TCP connection
```
Comment (sarenameas) [inline docs/doxygen/demos/http_demo.dox:43]: https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/docs/doxygen/demos/demos_subpages.dox#L14

File: docs/doxygen/demos/http_demo.dox [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+with a server, sends HTTP requests, then logs their responses.
+
+<p>
+An OpenSSL-based transport interface implementation is used to establish an encrypted
+TLS connection with both server and client authentication over port 443 to AWS IoT Core.
```
Comment (sarenameas) [inline docs/doxygen/demos/http_demo.dox:48]:
```suggestion
TLS connection, with both server and client authentication, over port 443 to AWS IoT Core.
```

**Review by sarenameas — DISMISSED** [review]

File: docs/doxygen/demos/http_demo.dox [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+/**
+@page http_demo coreHTTP
+@brief These demos demonstrate usage of the coreHTTP library.
+
+@section http_demo_plaintext HTTP Plaintext Demo
+@brief A demo of the coreHTTP library that establishes a plaintext TCP connection with a server, sends HTTP requests, then logs their responses.
+
+<p>
+A POSIX socket-based transport interface implementation is used to establish a
+plaintext connection over port 80 to an HTTP server. However, if a connection
```
Comment (aggarg) [inline docs/doxygen/demos/http_demo.dox:11]:
```suggestion
plaintext connection over port 80 to an HTTP server. If a connection 
```

File: docs/doxygen/demos/http_demo.dox [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+@image html http_demo_plaintext.png width=100%
+
+@section http_demo_basic_tls HTTP Basic TLS Demo
+@brief A demo of the coreHTTP library that establishes a TLS connection with server-only authentication,
+sends HTTP requests, then logs their responses.
+
+<p>
+An OpenSSL-based transport interface implementation is used to establish an encrypted
+TLS connection over port 443 to an HTTP server. Only the server's certificate
+is verified using its respective root CA certificate. However, if a connection
```
Comment (aggarg) [inline docs/doxygen/demos/http_demo.dox:30]:
```suggestion
is verified using its respective root CA certificate. If a connection
```

File: docs/doxygen/demos/http_demo.dox [hunk 8]
```diff
@@ -0,0 +1,65 @@
+
```
Comment (aggarg) [inline docs/doxygen/demos/http_demo.dox:1]: Nit: This file contains trailing spaces.

File: docs/doxygen/demos/mqtt_demo.dox [hunk 9]
```diff
@@ -1,49 +1,48 @@
 /**
-@page mqtt_demo_basic_tls MQTT Basic TLS Demo
+@page mqtt_demo coreMQTT
+@brief Demos used to illustrate various functionalities of MQTT libraries. They demonstrate the use of MQTT APIs to establish an MQTT session, subscribe to a topic filter, publish to a topic, receive incoming publishes, unsubscribe from a topic, and disconnect the MQTT session.
```
Comment (aggarg) [inline docs/doxygen/demos/mqtt_demo.dox:3]:
```suggestion
@brief Demos used to illustrate various functionalities of the MQTT Client library. They demonstrate the use of MQTT APIs to establish an MQTT session, subscribe to a topic filter, publish to a topic, receive incoming publishes, unsubscribe from a topic filter, and disconnect the MQTT session.
```

File: docs/doxygen/demos/mqtt_demo.dox [hunk 10]
```diff
@@ -1,49 +1,48 @@
...
 
 Messages in this demo are sent at QoS 2, which guarantees exactly one delivery according to the MQTT spec. See the demo workflow below:
 
 @image html mqtt_basic_tls.png width=100%
-*/
 
-/**
-@page mqtt_demo_mutual_auth MQTT Mutual Authentication Demo
+
+@section mqtt_demo_mutual_auth MQTT Mutual Authentication Demo
```
Comment (aggarg) [inline docs/doxygen/demos/mqtt_demo.dox:15]:
```suggestion
@section mqtt_demo_mutual_auth MQTT Mutual Auth Demo
```

**Review by aggarg — APPROVED** [review]

**Review by sarenameas — APPROVED** [review]

**Pull request merged**